Gems/MotionMatching/Code/Source/TrajectoryQuery.cpp

```diff
@@ -21,39 +21,100 @@ namespace EMotionFX::MotionMatching
         return displacement;
     }
 
+    void TrajectoryQuery::PredictFutureTrajectory(const ActorInstance* actorInstance,
+        const FeatureTrajectory* trajectoryFeature,
+        const AZ::Vector3& targetPos,
+        [[maybe_unused]] const AZ::Vector3& targetFacingDir)
+    {
+        const AZ::Vector3 actorInstanceWorldPosition = actorInstance->GetWorldSpaceTransform().m_position;
+        const AZ::Quaternion actorInstanceWorldRotation = actorInstance->GetWorldSpaceTransform().m_rotation;
+        const AZ::Vector3 actorInstanceToTarget = (targetPos - actorInstanceWorldPosition);
+
+        const size_t numFutureSamples = trajectoryFeature->GetNumFutureSamples();
+        const float numSections = aznumeric_cast<float>(numFutureSamples-1);
```

Comment:
nit: Is it worth making `numFutureSamples` signed and doing a `max(numFutureSamples-1, 0)`? (or assert `numFutureSamples` is always > `0`?)

Reply:
The trajectory feature itself makes sure there is at least one sample.